cryostatio/cryostat-web
# feat(router): routable views

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Fixes: #78 
Depends on: #925 (This PR includes refactoring to Settings and tab renaming).

## Description of the change:

Tab-based views are now routable using query paramter:

- [x] Recordings
- [x] Archives

Invalid tab query param will cause the view to fall back to the first tab. This way we don't have to handle error and show empty state and the user do not have to manually clean up url.

## Motivation for the change:

This allows users to bookmark urls for later without having to click on tabs. This would mean the user can use back-button to go back to previous (very nice UX if users have back-button on mouse).

## How to manually test:
1. `Run CRYOSTAT_IMAGE=quay.io... sh smoketest.sh...`
2. Go to any route with tab-based view (e.g. Recordings). Change tab and notice the query param changes.
3. Try bookmark/save the url with tab query param. Then visit it in another tab. Notice the correct tab is present.
